exporter/collector/logs.go

```diff
@@ -468,47 +444,52 @@ func (l logMapper) logToSplitEntries(
 		logRecord.Body().Map().PutStr(GCPTypeKey, GCPErrorReportingTypeValue)
 	}
 
-	entry.Labels = logLabels
-
 	// parse remaining OTel attributes to GCP labels
 	for k, v := range attrsMap {
 		// skip "gcp.*" attributes since we process these to other fields
 		if strings.HasPrefix(k, "gcp.") {
 			continue
 		}
-		if entry.Labels == nil {
-			entry.Labels = make(map[string]string)
-		}
 		if _, ok := entry.Labels[k]; !ok {
 			entry.Labels[k] = v.AsString()
 		}
 	}
 
-	// Calculate the size of the internal log entry so this overhead can be accounted
-	// for when determining the need to split based on payload size
-	// TODO(damemi): Find an appropriate estimated buffer to account for the LogSplit struct as well
-	logOverhead, err := l.logEntryToInternal(entry, logName, projectID, mr, 0, 0)
-	if err != nil {
-		return []logging.Entry{entry}, err
-	}
-	// make a copy so the proto initialization doesn't modify the original entry
-	overheadClone := proto.Clone(logOverhead)
-	overheadBytes := proto.Size(overheadClone)
-
-	payload, splits, err := parseEntryPayload(logRecord.Body(), l.maxEntrySize-overheadBytes)
-	if err != nil {
-		return []logging.Entry{entry}, err
+	if len(logRecord.Body().AsString()) == 0 {
+		return []*logpb.LogEntry{entry}, nil
 	}
 
-	// Split log entries with a string payload into fewer entries
-	if splits > 1 {
-		entries := make([]logging.Entry, splits)
-		payloadString := payload.(string)
-
+	switch logRecord.Body().Type() {
+	case pcommon.ValueTypeBytes:
+		s, err := toProtoStruct(logRecord.Body().Bytes().AsRaw())
+		if err != nil {
+			return nil, err
+		}
+		entry.Payload = &logpb.LogEntry_JsonPayload{JsonPayload: s}
+	case pcommon.ValueTypeMap:
+		s, err := structpb.NewStruct(logRecord.Body().Map().AsRaw())
+		if err != nil {
+			return nil, err
+		}
+		entry.Payload = &logpb.LogEntry_JsonPayload{JsonPayload: s}
+	case pcommon.ValueTypeStr:
+		// Calculate the size of the internal log entry so this overhead can be accounted
+		// for when determining the need to split based on payload size
+		// TODO(damemi): Find an appropriate estimated buffer to account for the LogSplit struct as well
+		overheadBytes := proto.Size(entry)
+		// Split log entries with a string payload into fewer entries
+		payloadString := logRecord.Body().AsString()
```

Comment:
nit since we've already checked its string typed
```suggestion
		payloadString := logRecord.Body().Str()
```